pkg/koordlet/statesinformer/states_pod.go

```diff
@@ -42,10 +49,22 @@ func (s *statesInformer) syncKubelet() error {
 	s.podMap = newPodMap
 	s.podHasSynced.Store(true)
 	s.podUpdatedTime = time.Now()
-	klog.Infof("get pods from kubelet success, len %d", len(s.podMap))
+	klog.Infof("get pods success, len %d", len(s.podMap))
 	return nil
 }
 
+func (s *statesInformer) getPodsFromApiserver() (corev1.PodList, error) {
+	timeoutInSecond := time.Second * time.Duration(s.config.ApiServerSyncTimeoutSeconds)
+	ctx, cancel := context.WithTimeout(context.Background(), timeoutInSecond)
+	defer cancel()
+	selector := fields.SelectorFromSet(fields.Set{"spec.nodeName": s.GetNode().Name})
+	opt := metav1.ListOptions{
+		FieldSelector: selector.String(),
```

Comment:
Consider adding `resourceVersion` and `resourceVersionMatch` to achieve better performance and scalability. See https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions for details.